zk/stages/stage_sequence_execute.go

```diff
@@ -47,6 +44,83 @@ func SpawnSequencingStage(
 		defer tx.Rollback()
 	}
 
+	lastBatch, err := stages.GetStageProgress(tx, stages.HighestSeenBatchNumber)
+	if err != nil {
+		return err
+	}
+
+	highestBatchInDS, err := cfg.datastreamServer.GetHighestBatchNumber()
+	if err != nil {
+		return err
+	}
+
+	if lastBatch >= highestBatchInDS {
```

Comment:
Is this check enough or it is possible to make it even more strict?

Not sure does if this makes sense, but for example:
- if lastBatch == highestBatchInDS + 1
- comparing parent hashes etc.

Reply:
Good question. I think the check will be sufficient for now, since the only case where the inconsistency can happen is when the sequencer is unwinded.